src/database.ts

```diff
@@ -976,6 +1009,20 @@ class Database extends GrpcServiceObject {
       callback!
     );
   }
+
+  async getRestoreInfo(): Promise<IRestoreInfoTranslatedEnum | undefined> {
+    const [metadata] = await this.getMetadata();
+    return metadata.restoreInfo ? metadata.restoreInfo : undefined;
+  }
+
+  async getState(): Promise<
+    | EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Database.State>
+    | undefined
+  > {
+    const [metadata] = await this.getMetadata();
+    return metadata.state ? metadata.state : undefined;
```

Comment:
Just `return metadata.state;` should have the same effect (I think?)

Reply:
Same here unfortunately :(

```
Type '"STATE_UNSPECIFIED" | "CREATING" | "READY" | "READY_OPTIMIZING" | null | undefined' is not assignable to type '"STATE_UNSPECIFIED" | "CREATING" | "READY" | "READY_OPTIMIZING" | undefined'.
```